src/searches.py

```diff
@@ -1,131 +1,187 @@
+            )
+            random.shuffle(trends)
+            for trend in trends:
+                self.googleTrendsShelf[trend] = None
+            logging.debug(
+                f"google_trends after load = {list(self.googleTrendsShelf.items())}"
+            )
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.googleTrendsShelf.__exit__(None, None, None)
+
+    def getGoogleTrends(self, wordsCount: int) -> list[str]:
+        # Function to retrieve Google Trends search terms
+        searchTerms: list[str] = []
+        i = 0
+        session = Utils.makeRequestsSession()
+        while len(searchTerms) < wordsCount:
+            i += 1
+            # Fetching daily trends from Google Trends API
+            r = session.get(
+                f"https://trends.google.com/trends/api/dailytrends?hl={self.browser.localeLang}"
+                f'&ed={(date.today() - timedelta(days=i)).strftime("%Y%m%d")}&geo={self.browser.localeGeo}&ns=15'
+            )
+            trends = json.loads(r.text[6:])
+            for topic in trends["default"]["trendingSearchesDays"][0][
+                "trendingSearches"
+            ]:
+                searchTerms.append(topic["title"]["query"].lower())
+                searchTerms.extend(
+                    relatedTopic["query"].lower()
+                    for relatedTopic in topic["relatedQueries"]
+                )
+            searchTerms = list(set(searchTerms))
+        del searchTerms[wordsCount : (len(searchTerms) + 1)]
+        return searchTerms
+
+    def getRelatedTerms(self, term: str) -> list[str]:
+        # Function to retrieve related terms from Bing API
+        relatedTerms: list[str] = requests.get(
+            f"https://api.bing.com/osjson.aspx?query={term}",
+            headers={"User-agent": self.browser.userAgent},
+        ).json()[1]
+        if not relatedTerms:
+            return [term]
+        return relatedTerms
+
+    def bingSearches(self) -> None:
+        # Function to perform Bing searches
+        logging.info(
+            f"[BING] Starting {self.browser.browserType.capitalize()} Edge Bing searches..."
+        )
+
+        self.browser.utils.goToSearch()
+
+        remainingSearches = self.browser.getRemainingSearches()
+        for searchCount in range(1, remainingSearches + 1):
+            # todo Disable cooldown for first 3 searches (Earning starts with your third search)
+            logging.info(f"[BING] {searchCount}/{remainingSearches}")
+            self.bingSearch()
+            time.sleep(random.randint(10, 15))
+
+        logging.info(
+            f"[BING] Finished {self.browser.browserType.capitalize()} Edge Bing searches !"
+        )
+
+    def bingSearch(self) -> None:
+        # Function to perform a single Bing search
+        pointsBefore = self.browser.utils.getAccountPoints()
+
+        rootTerm = list(self.googleTrendsShelf.keys())[1]
```

Comment:
Currently this yields:
```
2024-07-07 23:25:00,592 [ERROR]
Traceback (most recent call last):
  File "MS-Rewards-Farmer/main.py", line 40, in main
    earned_points = executeBot(currentAccount, args)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "MS-Rewards-Farmer/main.py", line 230, in executeBot
    searches.bingSearches()
  File "MS-Rewards-Farmer/src/searches.py", line 119, in bingSearches
    self.bingSearch()
  File "MS-Rewards-Farmer/src/searches.py", line 130, in bingSearch
    rootTerm = list(self.googleTrendsShelf.keys())[1]
               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^
```

Reply:
Try https://github.com/klept0/MS-Rewards-Farmer/pull/137#issuecomment-2198576551

Reply:
You ran out of trends. Delete the google_trends* files. I still think should increase the number of trends pulled, so this doesn't happen.

```
trends = self.getGoogleTrends(
                browser.getRemainingSearches(desktopAndMobile=True).getTotal() +10
            )
```

Reply:
Came across this myself, https://github.com/klept0/MS-Rewards-Farmer/pull/137#discussion_r1667877913 as a workaround fixed the immediate problem.

But, looks like the call to get trends is failing and causing `google_trends` to be empty:

> 2024-07-09 02:39:03,395 [DEBUG] google_trends = [('loadDate', datetime.date(2024, 7, 8)), ('wwe money in the bank', None)]
> 2024-07-09 02:39:11,524 [DEBUG] google_trends after load = [('loadDate', datetime.date(2024, 7, 9))]

Looking into it